compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -323,32 +318,41 @@ impl<'a: 'ast, 'ast, 'tcx> LateResolutionVisitor<'a, '_, 'ast, 'tcx> {
                 let mod_prefix =
                     mod_prefix.map_or_else(String::new, |res| (format!("{} ", res.descr())));
 
-                (mod_prefix, format!("`{}`", Segment::names_to_string(mod_path)), module_did, None)
+                (
+                    mod_prefix,
+                    &*format!("`{}`", Segment::names_to_string(mod_path)),
+                    module_did,
+                    None,
+                )
             };
 
-            let (fallback_label, suggestion) = if path_str == "async"
-                && expected.starts_with("struct")
-            {
-                ("`async` blocks are only allowed in Rust 2018 or later".to_string(), suggestion)
-            } else {
-                // check if we are in situation of typo like `True` instead of `true`.
-                let override_suggestion =
-                    if ["true", "false"].contains(&item_str.to_string().to_lowercase().as_str()) {
-                        let item_typo = item_str.to_string().to_lowercase();
-                        Some((item_span, "you may want to use a bool value instead", item_typo))
-                    // FIXME(vincenzopalazzo): make the check smarter,
-                    // and maybe expand with levenshtein distance checks
-                    } else if item_str.as_str() == "printf" {
-                        Some((
-                            item_span,
-                            "you may have meant to use the `print` macro",
-                            "print!".to_owned(),
-                        ))
-                    } else {
-                        suggestion
+            let (fallback_label, suggestion) =
+                if path_str == "async" && expected.starts_with("struct") {
+                    ("`async` blocks are only allowed in Rust 2018 or later".into(), suggestion)
+                } else {
+                    // check if we are in situation of typo like `True` instead of `true`.
+                    let override_suggestion = {
+                        let item_str_lower = item_str.as_str().to_lowercase();
+                        if ["true", "false"].contains(&item_str_lower.as_str()) {
+                            Some((
+                                item_span,
+                                "you may want to use a bool value instead",
+                                item_str_lower,
```

Comment:
```suggestion
                    let item_str_lower = item_str.as_str().to_lowercase();
                    let override_suggestion = if ["true", "false"].contains(&item_str_lower.as_str()) {
                            Some((
                                item_span,
                                "you may want to use a bool value instead",
                                item_str_lower,
                    // ...
                    };
```

Pull `item_str_lower` out to save an indentation level